File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        ListNode temp=head;
        int cnt=0;
        while(temp!=null){
            cnt++;
            temp=temp.next;
        }
        temp=head;
        int[] arr=new int[cnt];
        int i=0;
        while(temp!=null){
            arr[i++]=temp.val;
            temp=temp.next;
        }
        int total=0;
        int exp=0;
        for(int j=cnt-1;j>=0;j--){
            int p=(int)Math.pow(2,exp);
            int product=arr[j]*p;
            total+=product;
            exp++;
        }
        return total;
    }
}

File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        int cnt=0;
        ListNode* temp=head;
        while(temp!=NULL){
            cnt++;
            temp=temp->next;
        }
        int* arr=new int[cnt];
        temp=head;
        int i=0;
        while(temp!=NULL){
            arr[i++]=temp->val;
            temp=temp->next;
        }
        int total=0;
        int exp=0;
        for(int i=cnt-1;i>=0;i--){
            int p=pow(2,exp);
            int product=arr[i]*p;
            total+=product;
            exp++;
        }
        return total;
    }
};